akka-persistence-typed/src/main/scala/akka/persistence/typed/javadsl/PersistentBehavior.scala

```diff
@@ -169,11 +168,31 @@ abstract class PersistentBehavior[Command, Event, State >: Null] private (val pe
         })
       }).eventAdapter(eventAdapter())
 
-    if (supervisorStrategy.isDefined)
+    if (supervisorStrategy.isPresent)
       behavior.onPersistFailure(supervisorStrategy.get)
     else
       behavior
   }
 
 }
 
+/**
+ * A [[PersistentBehavior]] that is enforcing that replies to commands are not forgotten.
+ * There will be compilation errors if the returned effect isn't a [[ReplyEffect]], which can be
+ * created with `Effects().reply`, `Effects().noReply`, [[Effect.thenReply]], or [[Effect.thenNoReply]].
+ */
+@ApiMayChange
+abstract class PersistentBehaviorWithEnforcedReplies[Command, Event, State >: Null](persistenceId: PersistenceId, backoffSupervisorStrategy: Optional[BackoffSupervisorStrategy])
+  extends PersistentBehavior[Command, Event, State](persistenceId, backoffSupervisorStrategy) {
+
+  def this(persistenceId: PersistenceId) = {
+    this(persistenceId, Optional.empty[BackoffSupervisorStrategy])
+  }
+
+  def this(persistenceId: PersistenceId, backoffSupervisorStrategy: BackoffSupervisorStrategy) = {
+    this(persistenceId, Optional.ofNullable(backoffSupervisorStrategy))
+  }
+
+  // FIXME override commandHandler and commandHandlerBuilder to require the ReplyEffect return type,
+  // which is unfortunately intrusive to the CommandHandlerBuilder
```

Comment:
@renatocaval I continued implementing this more for real, in javadsl. It will require another type parameter to the CommandHandlerBuilder since it must use `Effect` or `ReplyEffect` as return type of the `matchCommand`. Makes it more complicated, but should be possible, just sharing progress.